src/main/java/com/yahoo/memory/StepBoolean.java

```diff
@@ -5,44 +5,28 @@
 
 package com.yahoo.memory;
 
-import java.util.concurrent.atomic.AtomicBoolean;
-
 /**
  * This is a step boolean function that can change its state only once and is thread-safe.
  *
  * @author Lee Rhodes
  */
 final class StepBoolean {
```

Comment:
You have defeated the purpose of this class.  Volatile is not guaranteed to be thread-safe, and what you propose is no longer a step-boolean as it can be changed back and forth.

Reply:
This class is used at just two places that are so simple that IMO it doesn't justify extra indirection and object allocation to "protect" from our own bugs in just a handful of lines of code.